spec/queries/homepage/articles_query_spec.rb

```diff
@@ -0,0 +1,101 @@
+require "rails_helper"
+
+RSpec.describe Homepage::ArticlesQuery, type: :query do
+  describe ".call" do
+    it "returns a relation object" do
+      expect(described_class.call).to be_a(ActiveRecord::Relation)
+    end
+
+    it "does not return draft articles" do
+      article = create(:article, published: false, published_at: nil)
+
+      expect(described_class.call.ids).not_to include(article.id)
+    end
+
+    it "returns both approved and unapproved articles by default" do
+      approved_article = create(:article, approved: true)
+      unapproved_article = create(:article, approved: false)
+
+      expected_result = [approved_article.id, unapproved_article.id]
+      expect(described_class.call.ids).to match_array(expected_result)
+    end
+
+    it "returns approved articles" do
+      article = create(:article, approved: true)
+
+      expect(described_class.call(approved: true).ids).to eq([article.id])
+    end
+
+    it "returns unapproved articles" do
+      article = create(:article, approved: false)
+
+      expect(described_class.call(approved: false).ids).to eq([article.id])
+    end
+
+    it "returns only published articles" do
+      article = create(:article)
+
+      expect(described_class.call.ids).to eq([article.id])
+    end
+
+    it "paginates by default" do
+      stub_const("Homepage::ArticlesQuery::DEFAULT_PER_PAGE", 1)
+
+      create_list(:article, 2)
+
+      expect(described_class.call.size).to eq(1)
+    end
+
+    it "supports pagination params" do
+      stub_const("Homepage::ArticlesQuery::DEFAULT_PER_PAGE", 1)
+
+      create_list(:article, 2)
+
+      expect(described_class.call(page: 1, per_page: 1).size).to eq(1)
+    end
+
+    it "filters by publication date", :aggregate_failures do
+      article = create(:article)
+
+      expect(described_class.call(published_at: article.published_at).size).to eq(1)
+      expect(described_class.call(published_at: (article.published_at - 1.month)..).size).to eq(1)
+      expect(described_class.call(published_at: (article.published_at + 1.month)).size).to eq(0)
```

Comment:
I'm finding this difficult to read, but that could just be because my neurons are deep-fried this week.

```suggestion
      expect(described_class.call(published_at: article.published_at).size).to eq(1)
      expect(described_class.call(published_at: 1.month.ago..).size).to eq(1)
      expect(described_class.call(published_at: 1.month.from_now)).to be_empty
```